app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    @property
+    def km(self) -> int | float:
+        return self._km
+
+    @km.setter
+    def km(self, value: int | float) -> None:
+        if not isinstance(value, (int | float)):
+            raise ValueError(
+                f"Unsupported type. <class 'float'> "
+                f"or <class 'int'> were expected. "
+                f"Got {type(value)} instead."
+            )
+        self._km = value
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (Distance, int, float)):
```

Comment:
you basically check for int and float in isinstance method twice. You can simplify this

Reply:
I don't get how I can do it with 1 ```return``` and 1 ```if``` cause if I get other as ```Distance``` it's gonna work fine, if I get other as ```int``` or ```float```, I'll have to make ```other = Distance(other)``` so I could compare their ```.km``` after.

Reply:
`    if isinstance(other, (int, float)):
        other = Distance(other)
    return Distance(self.km + other.km)` try following this pattern in your methods to simplify code

Reply:
> ` if isinstance(other, (int, float)): other = Distance(other) return Distance(self.km + other.km)` try following this pattern in your methods to simplify code

oh, right I didn't see it at first ;d